packages/diagrams/lib/src/future_builder.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+
+import 'diagram_step.dart';
+
+const String _futureBuilder = 'future_builder';
+const String _futureBuilderError = 'future_builder_error';
+const Duration _futureDuration = Duration(seconds: 1);
+const Duration _pauseDuration = Duration(seconds: 1);
+final Duration _totalDuration = _futureDuration + _pauseDuration;
+
+class FutureBuilderDiagram extends StatefulWidget implements DiagramMetadata {
+  const FutureBuilderDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  _FutureBuilderDiagramState createState() => _FutureBuilderDiagramState();
+}
+
+class _FutureBuilderDiagramState extends State<FutureBuilderDiagram> {
+  Future<String> _calculation;
+
+  @override
+  void initState() {
+    if (widget.name == _futureBuilder) {
+      _calculation = Future<String>.delayed(
+        _futureDuration,
+        () => 'Data Loaded',
+      );
+    } else if (widget.name == _futureBuilderError) {
+      _calculation = () async {
+        await Future<void>.delayed(_futureDuration);
+        throw 'Loading failed';
+      }();
+    }
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ConstrainedBox(
+      key: UniqueKey(),
+      constraints: BoxConstraints.tight(const Size(200, 120)),
+      child: Container(
+        alignment: FractionalOffset.center,
+        color: Colors.white,
+        child: FutureBuilder<String>(
+          future: _calculation,
+          builder: (BuildContext context, AsyncSnapshot<String> snapshot) {
+            switch (snapshot.connectionState) {
+              case ConnectionState.none:
+                return const Text('Press button to start.');
+              case ConnectionState.active:
+              case ConnectionState.waiting:
+                return const Text('Awaiting result...');
```

Comment:
This is a really cool way to explain and show off Futures. Well done!
Let's add a bit of visual fun here, how about an icon alongside this, like a clock?

Reply:
Sounds good! I've added a CircularProgressIndicator as the data loads and colorful Icons for Success / Error. Please let me know if ya like the updates!